gstreamer/src/event.rs

```diff
@@ -494,28 +501,50 @@ pub enum EventView<'a> {
     __NonExhaustive,
 }
 
-pub struct FlushStart<'a>(&'a EventRef);
+macro_rules! declare_concrete_event(
+    ($name:ident) => {
+        pub struct $name<'a> {
+            inner: &'a EventRef
+        }
+
+        impl<'a> $name<'a> {
+            pub fn new(inner: &'a EventRef) -> Self {
```

Comment:
This should not be public. Also why do you change it from a tuple strict? :)

Reply:
Note that .0 was private before too. You could only use it in the test because it is in a sub module

Reply:
I couldn't find another way to make the inner `EventRef` private.

Reply:
Ah OK! Will revert this then. :)